includes/reader-activation/class-reader-activation.php

```diff
@@ -2080,9 +2086,16 @@ public static function register_reader( $email, $display_name = '', $authenticat
 
 		$user_id = false;
 
-		if ( $existing_user && self::is_reader_without_password( $existing_user ) ) {
+		if ( $existing_user ) {
+			if ( ! self::is_user_reader( $existing_user ) ) {
+				self::send_non_reader_login_reminder( $existing_user );
+				return false;
+			}
+
 			// Don't send OTP email for newsletter signup.
```

Comment:
This comment should be updated to explain the new condition.

Reply:
Added an explanation in 458121d. Also made the email message more explicit in 8f96cc4:

> Someone attempted to sign into your account on *SITE_TITLE* using a login method reserved for reader accounts. Please use the following link to sign in via the WordPress admin dashboard:

Reply:
Thanks! WhatI meant was that the `Don't send OTP email for newsletter signup.` comment should be updated following the update of the condition, something like:

```suggestion
			// Don't send OTP email for newsletter signup, or if the reader has a password set. 
```